store/localcachelayer/channel_layer.go

```diff
@@ -150,6 +200,92 @@ func (s LocalCacheChannelStore) GetPinnedPostCount(channelId string, allowFromCa
 	return count, nil
 }
 
+// SaveMember is a wrapper method for the underlying store which just takes
+// care of the cache invalidation.
+func (s LocalCacheChannelStore) SaveMember(member *model.ChannelMember) (*model.ChannelMember, *model.AppError) {
+	defer s.invalidateMembersForUser(member.UserId)
+	return s.ChannelStore.SaveMember(member)
+}
+
+// UpdateMember is a wrapper method for the underlying store which just takes
+// care of the cache invalidation.
+func (s LocalCacheChannelStore) UpdateMember(member *model.ChannelMember) (*model.ChannelMember, *model.AppError) {
+	defer s.invalidateMembersForUser(member.UserId)
```

Comment:
Do I need to invalidate it when error happens? (same question for all the other cases).

Reply:
I think it's better we do. There are multiple queries that are executed in these methods, and not all of them happen in a transaction. So there is a chance that update gets executed, but a select query later fails, but we still need to invalidate it.